txnkv/txnlock/lock_resolver.go

```diff
@@ -693,6 +693,7 @@ func (lr *LockResolver) getTxnStatusFromLock(bo *retry.Backoffer, l *Lock, calle
 		// success before the primary region.
 		if err := bo.Backoff(retry.BoTxnNotFound, err); err != nil {
 			logutil.Logger(bo.GetCtx()).Warn("getTxnStatusFromLock backoff fail", zap.Error(err))
+			return TxnStatus{}, err
```

Comment:
I vaguely remember that this was intentionally designed in 4.0 to solve some problems, but I can't recall the details.
@MyonKeminta Do you remember?

Reply:
It seems to address some issues with resolving v4.0 optimistic mode, but it has no impact on the current default pessimistic mode. Maybe we still need to look into historical problems.